### PR TITLE
Add tests for arguments of correct type

### PR DESCRIPTION
**Pull request by aew**

### CODE REVIEW

File: scenarios/validation/ArgumentsOfCorrectType.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      query: |
+        query BadArgumentValue {
+          dog { name(surname: "notaboolean") }
+        }
+    when:
+      validate: [ArgumentsOfCorrectType]
+    then:
+      - error-count: 1
+      - error: "Argument 'surname' of type 'Boolean' has invalid value: \"notaboolean\". Reason: Boolean value expected."
+        loc: {line: 1, column: 30}
```
Comment (OlegIlyenko) [inline scenarios/validation/ArgumentsOfCorrectType.yaml]: I wonder where this location is is cumming from. As far as I can tell, the error is located here:

```
  dog { name(surname: "notaboolean") }
                      ^
```

So the location would be `{line: 2, column: 23}` in this case. Have I missed something?

Comment (aew) [reply scenarios/validation/ArgumentsOfCorrectType.yaml]: Updated this

**Comment by OlegIlyenko** [comment]
Thanks! and sorry for the delay.

I guess the error assertions are still ["under construction"](https://github.com/graphql-cats/graphql-cats/issues/5) - I tried to execute this in sangria but it produces slightly different error message :) I guess we need to eventually figure out the good error assertion format or agree on same error messages :)

Just out of curiosity, is this error message taken from particular GraphQL implementation library?

**Comment by aew** [comment]
Hi Oleg,

My bad for the lack of context.  I've been working on a really early stage implementation in Clojure, so that's where this is coming from.  And yes, the specific error message output is in flux (and it's a pretty high bar to have it identical to the character across platforms, but I don't have a better suggestion at the moment).

For further context, here are the test cases I used for validation:
https://github.com/tendant/graphql-clj/tree/master/test/scenarios/cats/validation

I don't think these are necessarily ready for a PR here, as I am still working on preserving line and column information from the parse after the tree visit.  But if you'd like WIP stuff like this to be part of this lib, I'd be happy to open a PR.

**Comment by OlegIlyenko** [comment]
Great! Thanks for sharing this info. Really glad to see the progress on GraphQL implementation for clojure!

Indeed, I also think that it would be hard to impose the same error messages for all of the implementations. I would like to find a nice solution for this, but unfortunately haven't got much time to focus on it recently. Hopefully after a few weeks I will get some time to work on it a bit more and maybe think of some format for it.

It's also really great that you are trying to write tests in CATs format! Would be happy to merge it. Also let us know whether you would be interested in joining the project and help us working on it (just ping me in [gitter chat](https://gitter.im/graphql-cats/graphql-cats)). I guess it will make it easier to contribute. 

We can also publish the scenario files to maven central or bintray (just package them as a JAR). I think it would be helpful in your case as well.

**Pull request merged**